tpv/core/entities.py

```diff
@@ -234,12 +234,17 @@ def evaluate_complex_property(self, prop, context, stringify=False):
         return self.process_complex_property(
             prop, context, lambda p, c: self.loader.eval_code_block(p, c, as_f_string=True), stringify=stringify)
 
+    def convert_env(self):
+        if isinstance(self.env, dict):
+            self.env = [dict(name=k, value=v) for (k, v) in self.env.items()]
+
     def validate(self):
         """
         Validates each code block and makes sure the code can be compiled.
         This process also results in the compiled code being cached by the loader,
         so that future evaluations are faster.
         """
+        self.convert_env()
```

Comment:
I think it would be better to move this to the constructor, at the point of first assignment? https://github.com/galaxyproject/total-perspective-vortex/blob/fadf3f671702a044e70b8a20ca302bf6f476db54/tpv/core/entities.py#L193